p2p/net/swarm/dial_worker.go

```diff
@@ -367,32 +350,25 @@ loop:
 // dispatches an error to a specific addr dial
 func (w *dialWorker) dispatchError(ad *addrDial, err error) {
 	ad.err = err
-	for _, reqno := range ad.requests {
-		pr, ok := w.pendingRequests[reqno]
-		if !ok {
-			// some other dial for this request succeeded before this one
-			continue
-		}
-
+	for pr := range w.pendingRequests {
```

Comment:
Same here.

Reply:
Yeah. Assuming we have two go routines waiting. 
1. We don't actually guarantee ordered delivery. This just publishes to a channel. This channel is being read [here](https://github.com/libp2p/go-libp2p/blob/master/p2p/net/swarm/dial_sync.go#L62). So the waiting go routines may be run in a different order.
2. This doesn't block. The channel has 1 capacity, so both the waiters are notified immediately. 

I don't know of any case where this might be an issue especially considering 1. 

Also [here](https://github.com/libp2p/go-libp2p/blob/master/p2p/net/swarm/dial_worker.go#L398) we do provide an out of order delivery. In this case the first requester will be informed of the connection after the second requester(which added a valid address) was informed.